File: packages/create-llama/templates/components/loaders/python/llama_parse/loader.py
import os
from llama_parse import LlamaParse
from llama_index.core import SimpleDirectoryReader

DATA_DIR = "data"  # directory containing the documents


def get_documents():
    if os.getenv("LLAMA_CLOUD_API_KEY") is None:
        raise ValueError(
            "LLAMA_CLOUD_API_KEY environment variable is not set. "
            "Please set it in .env file or in your shell environment then run again!"
        )
    parser = LlamaParse(result_type="markdown", verbose=True, language="en")

    reader = SimpleDirectoryReader(DATA_DIR, file_extractor={".pdf": parser})
    return reader.load_data()

File: packages/server/next/app/api/chat/engine/loader.ts
import {
  FILE_EXT_TO_READER,
  SimpleDirectoryReader,
} from "@llamaindex/readers/directory";

export const DATA_DIR = "./data";

export function getExtractors() {
  return FILE_EXT_TO_READER;
}

export async function getDocuments() {
  const documents = await new SimpleDirectoryReader().loadData({
    directoryPath: DATA_DIR,
  });
  // Set private=false to mark the document as public (required for filtering)
  for (const document of documents) {
    document.metadata = {
      ...document.metadata,
      private: "false",
    };
  }
  return documents;
}
